spec/langchain/assistants/assistant_spec.rb

```diff
@@ -50,4 +50,80 @@
       expect(thread.messages.last.content).to eq("bar")
     end
   end
+
+  describe "#run" do
+    let(:raw_openai_response) do
+      {
+        "id" => "chatcmpl-96QTYLFcp0haHHRhnqvTYL288357W",
+        "object" => "chat.completion",
+        "created" => 1711318768,
+        "model" => "gpt-3.5-turbo-0125",
+        "choices" => [
+          {
+            "index" => 0,
+            "message" => {
+              "role" => "assistant",
+              "content" => nil,
+              "tool_calls" => [
+                {
+                  "id" => "call_9TewGANaaIjzY31UCpAAGLeV",
+                  "type" => "function",
+                  "function" => {"name" => "calculator-execute", "arguments" => "{\"input\":\"2+2\"}"}
+                }
+              ]
+            },
+            "logprobs" => nil,
+            "finish_reason" => "tool_calls"
+          }
+        ],
+        "usage" => {"prompt_tokens" => 91, "completion_tokens" => 18, "total_tokens" => 109},
+        "system_fingerprint" => "fp_3bc1b5746b"
+      }
+    end
+
+    context "when auto_tool_execution is false" do
+      it "runs the assistant" do
+        allow(subject.llm).to receive(:chat).and_return(Langchain::LLM::OpenAIResponse.new(raw_openai_response))
```

Comment:
The `llm.chat()` method returns a response object so I wrapped the raw hash in that object.